File: JavaScript/8js 1/evalObject.js
/*
|--------------------------------------------------------------------------
| Switch/Case - Bug Fixing #6
|--------------------------------------------------------------------------
|
*/

function evalObject(value) {
   let result = 0;
   switch (value.operation) {
      case "+": result = value.a + value.b; break;
      case "-": result = value.a - value.b; break;
      case "/": result = value.a / value.b; break;
      case "*": result = value.a * value.b; break;
      case "%": result = value.a % value.b; break;
      case "^": result = Math.pow(value.a, value.b); break;
   }
   return result;
}


const evalObject = value => ({
   '+': (a, b) => a + b,
   '-': (a, b) => a - b,
   '/': (a, b) => a / b,
   '*': (a, b) => a * b,
   '%': (a, b) => a % b,
   '^': Math.pow,
}[value.operation](value.a, value.b));


function evalObject(value){
   switch(value.operation){
      case'+': return value.a + value.b;
      case'-': return value.a - value.b;
      case'/': return value.a / value.b;
      case'*': return value.a * value.b;
      case'%': return value.a % value.b;
      case'^': return Math.pow(value.a, value.b);
   }
}


console.log(evalObject({ a: 1, b: 1, operation: "+" }), 2);
console.log(evalObject({ a: 1, b: 1, operation: "-" }), 0);
console.log(evalObject({ a: 1, b: 1, operation: "/" }), 1);
console.log(evalObject({ a: 1, b: 1, operation: "*" }), 1);
console.log(evalObject({ a: 1, b: 1, operation: "%" }), 0);
console.log(evalObject({ a: 1, b: 1, operation: "^" }), 1);

File: Python/8py 1/eval_object.py
"""
|--------------------------------------------------------------------------
| Switch/Case - Bug Fixing #6
|--------------------------------------------------------------------------
|
"""

def eval_object(v):
   match v["operation"]:
      case "+": return v["a"] + v["b"]
      case "-": return v["a"] - v["b"]
      case "/": return v["a"] / v["b"]
      case "*": return v["a"] * v["b"]
      case "%": return v["a"] % v["b"]
      case "**": return v["a"] ** v["b"]
      case _: return 1


def eval_object(v):
   return eval("{a}{operation}{b}".format(**v))


def eval_object(v):
   return {"+": v['a']+v['b'],
         "-": v['a']-v['b'],
         "/": v['a']/v['b'],
         "*": v['a']*v['b'],
         "%": v['a']%v['b'],
         "**": v['a']**v['b'], }.get(v['operation'])
      

print(eval_object({'a':1,'b':1,'operation':'+'}), 2)
print(eval_object({'a':1,'b':1,'operation':'-'}), 0)
print(eval_object({'a':1,'b':1,'operation':'/'}), 1)
print(eval_object({'a':1,'b':1,'operation':'*'}), 1)
print(eval_object({'a':1,'b':1,'operation':'%'}), 0)
print(eval_object({'a':1,'b':1,'operation':'**'}), 1)